.github/workflows/CI_on_push.yml

```diff
@@ -0,0 +1,41 @@
+
+name: CI on push to main or issue-2 branch
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - issue-2-Setup_CI/CD_for_reference-models
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+
+      - name: Install clang and runtime dependencies
+        run: |
+          sudo apt install clang libunwind-dev
+          sudo apt install libgc-dev libre2-dev
+
+      - name: Checkout cli repo
+        uses: actions/checkout@v2
+        with:
+          repository: nervosum/nervosum
+          path: cli
+
+      - name: Build the CLI binary
+        run: |
+          cd cli
+          sbt cli/nativeLink
+
+      - name: Run CLI
+        run: cli/cli/target/scala-2.11/cli-out build
```

Comment:
This assumes that the build process uses `Scala 2.11` while this version is nowhere fixed. Can we somehow ensure that we control the scala version used during the build of cli, so that this step doesn't accidentally break in the future?

Reply:
I'd say let it break. It's easy to spot and fix. In the long run, we'll need to publish the artifacts and download them here instead anyway.

Reply:
I wouldn't necesserily go for the download binary approach, as this current approach serves as an integration test for neversum/nervosum#master. But other than that, yes, let's keep it as is for now. Indeed easy to spot/ fix.